agent/grpc-external/services/peerstream/stream_tracker.go

```diff
@@ -235,10 +248,19 @@ func (s *MutableStatus) TrackConnected() {
 	s.mu.Unlock()
 }
 
-func (s *MutableStatus) TrackDisconnected() {
+func (s *MutableStatus) TrackDisconnectedGracefully() {
+	s.mu.Lock()
+	s.Connected = false
+	s.DisconnectTime = s.timeNow().UTC()
+	s.DisconnectErrorMessage = ""
```

Comment:
Should `TrackConnected` also blank out `DisconnectErrorMessage`?

Reply:
Yes! Great catch.